### PR TITLE
feat(react): add `PhoneNumberInput` component

### PR DESCRIPTION
**Pull request by savindi7**
### Purpose

Add `PhoneNumberInput` component.

Screenshot 2023-03-10 at 17 54 46

### Related Issues
- https://github.com/wso2/oxygen-ui/issues/2

### Related PRs
- None

### Checklist
- [x] UX/UI review done on the final implementation.
- [x] Story provided. (Add screenshots)
- [x] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }}
+            {...SelectProps}
+          >
+            {countries?.map((country: Country) => (
+              <MenuItem value={country.dial_code} className="oxygen-dial-code-menu-item">
+                <ListItemIcon>
+                  <Image
+                    loading="lazy"
+                    src={`https://flagcdn.com/${country.code.toLowerCase()}.svg`}
+                    srcSet={`https://flagcdn.com/${country.code.toLowerCase()}.svg 2x`}
```
Comment (thivi) [inline packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx:95]: Using a CDN in a library does not seem right to me. Can we try using flag emojis?
https://dev.to/jorik/country-code-to-flag-emoji-a21

Comment (brionmario) [reply packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx:95]: Emojis might have some issues in Windows. Maybe we can try a SVG library.

Comment (savindi7) [reply packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx]: Added [react-world-flags](https://www.npmjs.com/package/react-world-flags?activeTab=readme) package

File: packages/react/src/components/PhoneNumberInput/constants/countries.ts [hunk 2]
```diff
@@ -0,0 +1,1236 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export interface Country {
+  code: string;
```
Comment (brionmario) [inline packages/react/src/components/PhoneNumberInput/constants/countries.ts:20]: Lets document the interface.

File: packages/react/src/components/PhoneNumberInput/constants/countries.ts [hunk 3]
```diff
@@ -0,0 +1,1236 @@
...
+  dialCode: string;
```
Comment (brionmario) [inline packages/react/src/components/PhoneNumberInput/constants/countries.ts:21]: Is dialcode a common term?

File: packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        onChange(`${country.dialCode}${phoneNumber}`);
+      }
+    }, [country, phoneNumber, onChange]);
+
+    const handleCountryCodeChange = (event: SelectChangeEvent): void => {
+      setCountry(countries.find((item: Country) => item.dialCode === event.target.value));
+    };
+
+    const handlePhoneNumberChange = (event: ChangeEvent<HTMLInputElement>): void => {
+      setPhoneNumber(event.target.value);
```
Comment (brionmario) [inline packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx:83]: Shall we remove the useEffect and emit the values on the two onChange event handlers?
```jsx
*interface*
onChange(countryCode, phoneNumber)

...
      setPhoneNumber(event.target.value);
      onChange(country, event.target.value)
```

File: packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import Box, {BoxProps} from '../Box';
+import InputLabel from '../InputLabel';
+import ListItemIcon from '../ListItemIcon';
+import MenuItem from '../MenuItem';
+import './phone-number-input.scss';
+import OutlinedInput, {OutlinedInputProps as MuiOutlinedInputProps} from '../OutlinedInput';
+import Typography from '../Typography';
+
+export interface PhoneNumberInputProps extends BoxProps {
+  OutlinedInputProps?: Omit<MuiOutlinedInputProps, 'id' | 'label' | 'placeholder' | 'value' | 'type'>;
```
Comment (brionmario) [inline packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx:44]: Lets document the interface.

Comment (savindi7) [reply packages/react/src/components/PhoneNumberInput/PhoneNumberInput.tsx:41]: Fixed https://github.com/wso2/oxygen-ui/pull/64/commits/481bd8618800338377fcff1956b262066106c9c5

File: packages/react/src/components/PhoneNumberInput/constants/countries.ts (hunk 2 above)
Comment (savindi7) [reply packages/react/src/components/PhoneNumberInput/constants/countries.ts:25]: Fixed https://github.com/wso2/oxygen-ui/pull/64/commits/481bd8618800338377fcff1956b262066106c9c5

File: packages/react/src/components/PhoneNumberInput/constants/countries.ts (hunk 3 above)
Comment (brionmario) [reply packages/react/src/components/PhoneNumberInput/constants/countries.ts:30]: IMO, country code is the most commonly used term.

https://countrycode.org/

**Review by brionmario — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

File: packages/react/package.json [hunk 6]
```diff
@@ -46,7 +46,9 @@
     "@mui/utils": "^5.10.16",
     "@oxygen-ui/primitives": "*",
     "@oxygen-ui/react-icons": "*",
-    "clsx": "^1.2.1"
+    "@storybook/addon-essentials": "^6.5.16",
```
Comment (thivi) [inline packages/react/package.json:49]: Shouldn't this be a dev dependency?

File: packages/react/src/components/PhoneNumberInput/constants/countries.ts (hunk 3 above)
Comment (thivi) [reply packages/react/src/components/PhoneNumberInput/constants/countries.ts:30]: Country Calling Codes or Country Dial-in Codes seems to be the standard. 
https://en.wikipedia.org/wiki/List_of_country_calling_codes

File: packages/react/package.json (hunk 6 above)
Comment (savindi7) [reply packages/react/package.json:49]: Fixed with - https://github.com/wso2/oxygen-ui/pull/66

**Pull request merged**